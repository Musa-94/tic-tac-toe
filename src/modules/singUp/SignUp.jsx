import React from 'react';
import CustomButton from 'src/components/customButton';
import { withLocalization } from 'src/lang/localization';
import './signUpStyles.less';

const SignUp = props => {
    const {
        t
    } = props;

    return (
        <div className={'wrapper'}>
            <div className={'container__header'}
                 children={t('registration')}
            />
            <div className={'container__content'}>
                <form className={'content'}>
                    <input type={'text'}
                           title={'Введите логин'}
                           required
                           className={'login input'}
                           placeholder={t('enterLogin')}
                    />
                    <input type={'password'}
                           title={'Введите пароль'}
                           required
                           className={'input'}
                           placeholder={t('enterPassword')}
                    />
                    <input type={'password'}
                           title={'Введите подтвержденние пароля'}
                           required
                           className={'input'}
                           placeholder={t('confirmThePassword')}
                    />
                    <CustomButton
                        textTitle={t('signUp')}
                    />
                </form>
            </div>
        </div>
    )
}

export default React.memo(withLocalization()(SignUp));
