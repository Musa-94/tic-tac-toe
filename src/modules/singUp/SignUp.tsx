import React from 'react';
import CustomButton from '../../components/customButton';
import { SingUpProps } from '../../types';
import { withLocalization } from '../../lang/localization';
import './signUpStyles.less';
// import { ThemeProvider } from 'styled-components';
import { Content,
         Container,
         SignUpForm,
         WrapperSignUp } from './styledComponents';

const SignUp: React.FC<SingUpProps> = (props: SingUpProps): React.ReactElement => {
    const {
        t
    } = props;

    return (
            <WrapperSignUp data-at={'wrapper'}>
                <Container.header data-at={'container__header'}
                     children={t('registration')}
                />
                <Content data-at={'container__content'}>
                    <SignUpForm data-at={'content'}>
                        <SignUpForm.input type={'text'}
                               title={t('enterLogin')}
                               required
                               className={'login input'}
                               placeholder={t('enterLogin')}
                        />
                        <SignUpForm.input type={'password'}
                               title={t('enterPassword')}
                               required
                               className={'input'}
                               placeholder={t('enterPassword')}
                        />
                        <SignUpForm.input type={'password'}
                               title={t('confirmThePassword')}
                               required
                               className={'input'}
                               placeholder={t('confirmThePassword')}
                        />
                        <CustomButton
                            textTitle={t('signUp')}
                        />
                    </SignUpForm>
                </Content>
            </WrapperSignUp>
    );
}

export default React.memo(withLocalization()(SignUp));
