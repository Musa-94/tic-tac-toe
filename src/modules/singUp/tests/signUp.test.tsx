import React from 'react';
import SignUp from '../SignUp';

jest.mock('react-i18next')

const props = {
    t: () => '',
}

describe('SignUp snapshot', () => {
    it('should correctly render', () => {
        const component = renderer(<SignUp {...props}/>);

        expect(component).toMatchSnapshot();
    });
});
