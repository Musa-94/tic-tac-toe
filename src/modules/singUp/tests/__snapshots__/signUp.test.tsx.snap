// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp snapshot should correctly render 1`] = `
<div
  className="wrapper"
>
  <div
    className="container__header"
  >
    
  </div>
  <div
    className="container__content"
  >
    <form
      className="content"
    >
      <input
        className="login input"
        placeholder=""
        required={true}
        title=""
        type="text"
      />
      <input
        className="input"
        placeholder=""
        required={true}
        title=""
        type="password"
      />
      <input
        className="input"
        placeholder=""
        required={true}
        title=""
        type="password"
      />
      <button
        className="border-button"
        type="submit"
      >
        
      </button>
    </form>
  </div>
</div>
`;
